torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -853,28 +853,41 @@ bool AliasDb::mayContainAlias(
     const at::ArrayRef<Value*>& a,
     const at::ArrayRef<Value*>& b) const {
   std::vector<Element*> a_elements;
+  bool a_cannot_check_containment = false;
   for (const auto& val : a) {
     if (cannotCheckAliasContainment(val)) {
-      return true;
+      a_cannot_check_containment = true;
```

Comment:
nit: I think it might be a bit clearer if we just do `return b_elements.size() == 0` with a comment here instead of keeping a variable alive.

Reply:
that doesn't cover the case where b contains immutable elements

Reply:
`b_elements.size() == 0` will always be true because we haven't iterated over `b` yet

Reply:
mmm makes sense lol